src/ray/rpc/worker/core_worker_client.cc

```diff
@@ -0,0 +1,134 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/rpc/worker/core_worker_client.h"
+
+#include <limits>
+#include <memory>
+#include <utility>
+
+namespace ray {
+namespace rpc {
+
+CoreWorkerClient::CoreWorkerClient(
+    const rpc::Address &address,
+    ClientCallManager &client_call_manager,
+    std::function<void()> core_worker_unavailable_timeout_callback)
+    : addr_(address) {
```

Comment:
nit (no need to change): it's recommended to pass `address` by value, and move to `addr_`
Reference: modern effective C++ item-41

Reply:
```cpp
    rpc::Address address,
    ClientCallManager &client_call_manager,
    std::function<void()> core_worker_unavailable_timeout_callback)
    : addr_(std::move(address)) {
```